piIOComm.c

```diff
@@ -50,176 +51,11 @@
-			pr_info("UartThread: received TERM signal\n");
-			break;
-		} else if (r != MAX_READ_BUF) {	// not finished yet
-			clear();
-			return -1;
-		} else {
-			down(&recvLenSem);
-			if (i16uRecvLen_s > 0) {
-				enqueue(acBuf_l[0]);
-				i16uRecvLen_s--;
-				if (i16uRecvLen_s < REV_PI_RECV_IO_HEADER_LEN && i16uRecvLen_s >= REV_PI_RECV_IO_HEADER_LEN - IOPROTOCOL_HEADER_LENGTH) {
-					// if piIoComm_recv was called with the length value REV_PI_RECV_IO_HEADER_LEN,
-					// the length in the received header is used.
-					int l = REV_PI_RECV_IO_HEADER_LEN - i16uRecvLen_s;
-					ioHeader_l.ai8uHeader[l - 1] = acBuf_l[0];
-					pr_info("UartThread: %d %02x\n", l, acBuf_l[0]);
-					if (l == IOPROTOCOL_HEADER_LENGTH) {
-						// now we have received two bytes and can read length field
-						// we already received the header, set the length to the length of data plus crc byte
-						i16uRecvLen_s = ioHeader_l.sHeaderTyp1.bitLength + 1;
-						pr_info("UartThread: len=%d\n", i16uRecvLen_s);
-					}
-				}
-				if (i16uRecvLen_s == 0) {
-					up(&queueSem);
-				}
-			} else {
-				enqueue(acBuf_l[0]);
-			}
-			up(&recvLenSem);
-		}
-	}
-
-	pr_info("UART Thread Exit\n");
-
-	return (0);
-}
-
-int piIoComm_open_serial(void)
-{
-	/*
-	 * Oeffnet seriellen Port
-	 * Gibt das Filehandle zurueck oder -1 bei Fehler
-	 *
-	 * RS232-Parameter:
-	 * 115200 bps, 8 Datenbits, 1 Stoppbit, even parity, no handshake
-	 */
-
-	struct file *fd;	/* Filedeskriptor */
-	struct termios newtio;	/* Schnittstellenoptionen */
-
-	/* Port oeffnen - read/write, kein "controlling tty", Status von DCD ignorieren */
-	fd = filp_open(REV_PI_TTY_DEVICE, O_RDWR | O_NOCTTY, 0);
-	if (!IS_ERR_OR_NULL(fd)) {
-		int r;
-		mm_segment_t oldfs;
-
-		oldfs = get_fs();
-		set_fs(KERNEL_DS);
-
-		/* get the current options */
-		r = fd->f_op->unlocked_ioctl(fd, TCGETS, (unsigned long)&newtio);
-		if (r < 0) {
-			set_fs(oldfs);
-			pr_info("unlocked_ioctl TCGETS failed %d\n", r);
-			return (-1);
-		}
-#if 0
-		newtio.c_iflag &= ~(IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR | IGNCR | ICRNL | IXON);
-		newtio.c_oflag &= ~OPOST;
-		newtio.c_lflag &= ~(ECHO | ECHONL | ICANON | ISIG | IEXTEN);
-		newtio.c_cflag &= ~(CSIZE);
-		newtio.c_cflag |= CS8 | PARENB | B115200;
-#else
-		newtio.c_iflag = 0;
-		newtio.c_oflag = 0;
-		newtio.c_lflag = 0;
-		newtio.c_cflag = CLOCAL | CREAD;
-		newtio.c_cflag |= CS8 | PARENB | B115200;
-#endif
-		if (fd->f_op->unlocked_ioctl(fd, TCSETS, (unsigned long)&newtio) < 0) {
-			set_fs(oldfs);
-			pr_info("unlocked_ioctl TCSETS failed\n");
-			return -1;
-		}
-		set_fs(oldfs);
-	} else {
-		pr_err("could not open device %s", REV_PI_TTY_DEVICE);
-		return -1;
-	}
-	piIoComm_fd_m = fd;
-
-	pr_info_serial("filp_open %d\n", (int)piIoComm_fd_m);
-
-	sema_init(&queueSem, 0);
-	sema_init(&recvLenSem, 1);
-	clear();
-
-	return 0;
-}
-
 int piIoComm_send(INT8U * buf_p, INT16U i16uLen_p)
 {
-	ssize_t write_l = 0;
-	INT16U i16uSent_l = 0;
+	int written;
 
 #ifdef DEBUG_SERIALCOMM
```

Comment:
this code block is also useless anymore

Reply:
Why is that? It still provides the same functionality as before...

Reply:
That variable is not used in that code block. Only i16uLen_p is used which still exists

Reply:
Oh, yes, sorry, i16uLen_p  vs i16uSent_l  I made mistake with them.